clients/src/main/java/org/apache/kafka/common/Node.java

```diff
@@ -100,12 +101,15 @@ public String rack() {
 
     @Override
     public int hashCode() {
+        if (hash != 0)
+            return hash;
```

Comment:
if all the fields in this class are `final`, then why not compute `hash` in the constructor and just return it here?

Reply:
You don't want to pay the cost for the computation of the hashCode if it's never used. I still think it would be better to make the hashCode very cheap by just using the `id`.